internal/dao/dp.go

```diff
@@ -34,6 +34,28 @@ type Deployment struct {
 	Resource
 }
 
+func (d *Deployment) Create(ctx context.Context, ns string, obj runtime.Object) (runtime.Object, error) {
+	dp, ok := obj.(*appsv1.Deployment)
+	if !ok {
+		return nil, fmt.Errorf("Expecting *appsv1.Deployment but got %T", obj)
+	}
+
+	auth, err := d.Client().CanI(ns, "apps/v1/deployments:create", []string{client.CreateVerb})
```

Comment:
Isn't it better to try creating the resource and report auth errors in case? This adds an extra - unnecessary - api call to `CanI`.